### PR TITLE
discord workflow

### LINKED ISSUE #292: Notify Discord Community Channel of pre-releases
📋 [#prerelease-testing discord channel](https://discord.com/channels/864285291518361610/999388864063995914)

In an effort to help notify the community of our pre-releases, I think we should explore ways to automate discord messages when we make pre-releases, perhaps via [github actions like this](https://github.com/marketplace/actions/actions-for-discord).

### PR DESCRIPTION
**Pull request by ozgunozerk**
Outcome: 
image

Tested on `discord-testing` channel. 
Nazar provided the webhook for the `prelease-testing` channel, and updated the secret in the repo, so we should be good to go. 

This workflow will be triggered when `publish` is completed, and the jobs will run ONLY IF `publish` is successful.

The message itself is probably not the best, any edit/suggestion is welcomed

---

closes #292

### CODE REVIEW

**Review by nazar-pc — COMMENTED** [review]
Should this be a separate workflow or only be published once all builds are finished? You can make job depend on other jobs.

**Comment by ozgunozerk** [comment]
Yes I was aware of I can make jobs depend on others, but went with to make workflows depend on others :)

We chose to make it a separate workflow in the desktop meeting (with Justin, Arthur and Serge). I'm not against both options if there is a strong argument for any of them. Copying these and including on the bottom of the publish will not be much extra work for me. But I also prefer this way.

**Review by nazar-pc — APPROVED** [review]

**Review by achiurizo — APPROVED** [review]

**Comment by achiurizo** [comment]
image


Let's update the bot name to something like  `Subspace Desktop Releases` and use the Subspace bot icon.

Also nitpick but could simplify our messaging like this(note the discord text link markdown):

```
Subspace Desktop [0.6.11](https://github.com/subspace/subspace-desktop/releases/tag/0.6.11) has been released! Check it out on the [releases page](https://github.com/subspace/subspace-desktop/releases).
```

which should render something like

> Subspace Desktop [0.6.11](https://github.com/subspace/subspace-desktop/releases/tag/0.6.11) has been released! Check it out on the [releases page](https://github.com/subspace/subspace-desktop/releases).

File: .github/workflows/discord.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        id: extract_version
+        uses: Saionaro/extract-package-version@v1.0.6
+
+      - name: Discord notification
+        env:
+          DISCORD_USERNAME: Subspace_Desktop-Releaser
+          DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK }}
+        uses: Ilshidur/action-discord@master
+        with:
+          args: "Subspace desktop version: ${{ steps.extract_version.outputs.version }} has been released! Check it out in this link: https://github.com/subspace/subspace-desktop/releases/tag/${{ steps.extract_version.outputs.version }}"
```
Comment (achiurizo) [inline .github/workflows/discord.yml:26]:
```suggestion
          args: "Subspace Desktop [${{ steps.extract_version.outputs.version}}](https://github.com/subspace/subspace-desktop/releases/tag/${{ steps.extract_version.outputs.version}}) has been released! Check it out on the [releases page](https://github.com/subspace/subspace-desktop/releases)."
```

File: .github/workflows/discord.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Checkout code
+        uses: actions/checkout@master
+
+      - name: Extract version
+        id: extract_version
+        uses: Saionaro/extract-package-version@v1.0.6
+
+      - name: Discord notification
+        env:
+          DISCORD_USERNAME: Subspace_Desktop-Releaser
```
Comment (achiurizo) [inline .github/workflows/discord.yml:22]:
```suggestion
          DISCORD_USERNAME: "Subspace Desktop Releases"
```
does this work?

Also could add `DISCORD_AVATAR` to add our Subspace Bot Icon.

**Comment by ozgunozerk** [comment]
Oh... Did not think you would have more suggestions @achiurizo , in such cases, you can disable the auto-merge if you like, no problem for me :)
I will submit another PR about suggestions today

**Pull request merged**